### PR TITLE
Improve documentation and upload

### PR DESCRIPTION
**Pull request by percyliang**
- Change 'cl upload dataset/program file' to 'cl upload file'
- Change 'cl ls worksheet' to 'cl ls -w worksheet'
- Add 'cl help -v' to show all help
- Added documentation for commands and made it more consistent
- When uploading bundle with weird name, gets sanitized to default name instead of throwing an error
- Fix quoting bug in copying files
  @kashizui please review

### CODE REVIEW

**Comment by percyliang** [comment]
Fixes #212 #152 #21

File: codalab/lib/bundle_cli.py [hunk 1]
```diff
@@ -695,11 +734,9 @@ def do_upload_command(self, args):
             raise UsageError('Nothing to upload')
 
         # Pull out the upload bundle type from the arguments and validate it.
-        if args.bundle_type not in UPLOADED_TYPES:
-            raise UsageError('Invalid bundle type %s (options: [%s])' % (
-              args.bundle_type, '|'.join(sorted(UPLOADED_TYPES)),
-            ))
-        bundle_subclass = get_bundle_subclass(args.bundle_type)
+        # Note: only allow dataset bundles (eventually deprecate the program bundle and just have )
+        bundle_type = 'dataset'
```
Comment (kashizui) [inline codalab/lib/bundle_cli.py:738]: Should we potentially just set this to `'uploaded'` (and add the appropriate fields in `UploadedBundle` to make it a concrete type)?

**Comment by kashizui** [comment]
LGTM!

File: codalab/lib/bundle_cli.py (hunk 1 above)
Comment (percyliang) [inline codalab/lib/bundle_cli.py:738]: Eventually, but for now, everything in the system is a dataset or program
bundle_type, so let's leave it that way to not break anything.

**Pull request merged**